conformance/tests/overloads_evaluation.py

```diff
@@ -176,49 +211,126 @@ def check_expand_tuple(v: int | str) -> None:
 # > that supplies an indeterminate number of positional or keyword arguments.
 # > If so, eliminate overloads that do not have a variadic parameter.
 
+
 @overload
-def variadic(x: int, /) -> str:
-    ...
+def variadic(x: int, /) -> str: ...
+
 
 @overload
-def variadic(x: int, y: int, /, *args: int) -> int:
-    ...
+def variadic(x: int, y: int, /, *args: int) -> int: ...
+
 
 def variadic(*args: int) -> int | str:
     return 1
 
+
 # > - If this results in only one remaining candidate overload, it is
 # >   the winning match. Evaluate it as if it were a non-overloaded function
 # >   call and stop.
 
+
 def check_variadic(v: list[int]) -> None:
     ret1 = variadic(*v)
     assert_type(ret1, int)
 
 
-# > Step 5: For each argument, determine whether all possible
+# > Step 5: For all arguments, determine whether all possible
 # > :term:`materializations <materialize>` of the argument's type are assignable to
 # > the corresponding parameter type for each of the remaining overloads. If so,
 # > eliminate all of the subsequent remaining overloads.
 
+
 @overload
-def example4(x: list[int], y: int) -> int:
-   ...
+def example4(x: list[int], y: int) -> list[int]: ...
+
 
 @overload
-def example4(x: list[str], y: str) -> int:
-    ...
+def example4(x: list[str], y: str) -> list[int]: ...
+
 
 @overload
-def example4(x: int, y: int) -> list[int]:
-    ...
+def example4(x: int, y: int) -> list[str]: ...
+
+
+def example4(x: list[int] | list[str] | int, y: int | str) -> list[int] | list[str]:
+    return []
 
-def example4(x: list[int] | list[str] | int, y: int | str) -> int | list[int]:
-    return 1
 
-def check_example4(v1: list[Any], v2: Any):
+def check_example4(v1: list[Any], v2: Any) -> None:
     ret1 = example4(v1, v2)
-    assert_type(ret1, int)
+    assert_type(ret1, list[int])
 
     ret2 = example4(v2, 1)
     assert_type(ret2, Any)
+
+
+@overload
+def example5(obj: list[int]) -> list[int]: ...
+
+
+@overload
+def example5(obj: list[str]) -> list[str]: ...
+
+
+def example5(obj: Any) -> list[Any]:
+    return []
+
+
+def check_example5(b: list[Any]) -> None:
+    assert_type(example5(b), Any)
+
+
+@overload
+def example6(a: int, b: Any) -> float: ...
+
+
+@overload
+def example6(a: float, b: T) -> T: ...
+
+
+def example6(a: float, b: T) -> T: ...
+
+
+def check_example6(a: list, b: Any) -> None:
+    m: list[int] = []
+
+    v1 = example6(1, a)
+    assert_type(v1, float)
+
+    v2 = example6(1, b)
+    assert_type(v2, float)
+
+    v3 = example6(1, m)
+    assert_type(v3, float)
+
+    v4 = example6(1.0, "")
+    assert_type(v4, str)
```

Comment:
It seems like an unwarranted assumption here that this must be `str` and not `Literal[""]` -- is that literal-widening behavior of typevar constraint solving already mandated elsewhere in the spec?

If we want to avoid that issue here (which seems wise to me if it is not already clearly mandated), we could use an argument to `check_example6` typed as `str`, instead of using `""`.